pyvrp/cpp/CostEvaluator.h

```diff
@@ -18,11 +18,22 @@ template <typename T> concept CostEvaluatable = requires(T arg)
     { arg.excessLoad() } -> std::same_as<Load>;
     { arg.fixedVehicleCost() }  -> std::same_as<Cost>;
     { arg.timeWarp() } -> std::same_as<Duration>;
-    { arg.uncollectedPrizes() } -> std::same_as<Cost>;
+    { arg.empty() } -> std::same_as<bool>;
     { arg.isFeasible() } -> std::same_as<bool>;
     // clang-format on
 };
 
+// If, additionally, methods related to optional clients and prize collecting
+// are implemented we can also take that aspect into account. See the
+// CostEvaluator implementation for details.
+template <typename T>
+concept PrizeCostEvaluatable = CostEvaluatable<T> and requires(T arg)
```

Comment:
I don't see how these mean different things. For both a route and a whole solution, this is the fixed vehicle cost of all assigned vehicles. That cost is only counted when the object is non-empty, which in the case of a solution is further constrained because we do not allow empty routes there. But that does not change the meaning IMO.

Reply:
Sorry I was confused here. Since we don't allow empty routes in a solution it is indeed consistent. :)